packager/mpd/base/xml/xml_node.cc

```diff
@@ -427,6 +431,46 @@ bool RepresentationXmlNode::AddVODOnlyInfo(const MediaInfo& media_info) {
     RCHECK(AddChild(std::move(child)));
   }
 
+  if (use_segment_list) {
+    XmlNode child("SegmentList");
+    if (media_info.has_reference_time_scale()) {
+      RCHECK(child.SetIntegerAttribute("timescale",
+                                       media_info.reference_time_scale()));
+    }
+
+    if (media_info.has_reference_time_scale()) {
+      RCHECK(child.SetIntegerAttribute("timescale",
+                                       media_info.reference_time_scale()));
+
+      if (media_info.has_media_duration_seconds()) {
+        const uint64_t duration_seconds =
+            (uint64_t) (target_segment_duration * media_info.reference_time_scale());
```

Comment:
Is the explicit cast necessary? Also, we use C++ style cast in this project if needed, i.e.

static_cast<uint64_t>(...)